common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -1028,6 +1028,12 @@
           "cannot find a static method <methodName> that matches the argument types in <className>."
         ]
       },
+      "UNSUPPORTED_CASE_WHEN_CLAUSE" : {
+        "message" : [
+          "The CASE WHEN clause of type <clause> is not supported."
+        ],
+        "sqlState" : "42601"
+      },
```

Comment:
I think this should probably be vice-versa: i.e. `UNSUPPORTED_LAMBDA_USAGE` or something along those lines

Reply:
I pinged @srielau regarding this one so we can think of error name that suites this case the most.

Reply:
I think we can use stronger verbiage than unsupported. I dare say that we will "never" support a lambda function outside of an argument to a dedicated function consuming them.
INVALID_LAMBDA_USAGE "42K0E"
The lambda function <lambdaExpr> cannot be used in this context Lambda functions must be passed as arguments to functions defined to accept them, such as array_sort.

Reply:
Done, thanks!